dashboard/server.R

```diff
@@ -104,61 +105,220 @@ function(input, output, session) {
              xaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE),
              yaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE))
-    
   })
   
-  # Bach donut
+   # Bach donut
+  colors_education <- c("#1b587c", "#9f2936", "#f2f2f2")
+  
   output$bach_donut <- renderPlotly({
     refine_top_20() %>%
-      mutate(Bach_or_higher = ifelse(is.na(Bach_or_higher), "No response", Bach_or_higher)) %>%
+      mutate(Bach_or_higher = ifelse(is.na(Bach_or_higher), "No Response", Bach_or_higher),
+             Bach_or_higher = factor(Bach_or_higher, levels = c("No", "Yes", "No Response"))) %>%
       group_by(Bach_or_higher) %>%
       summarize(count = n()) %>%
-      plot_ly(labels = ~Bach_or_higher, values = ~count) %>%
+      plot_ly(labels = ~Bach_or_higher, values = ~count,
+              textinfo = "label", # "label+percent"
+              textfont = list(size = 10),
+              marker = list(colors = colors_education)) %>%
       add_pie(hole = 0.5) %>%
       layout(title = "Bachelor's or higher",  
-             showlegend = TRUE,
+             showlegend = FALSE,
              legend = list(orientation = "h"),
              xaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE),
              yaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE))
-    
   })
   
+  
   #  Donut for white/non-white
+  
+  colors_white <- c("#1b587c", "#9f2936")
   output$race_donut <- renderPlotly({
     refine_top_20() %>%
       # mutate(non_white = ifelse(is.na(Bach_or_higher), "No response", Bach_or_higher)) %>%
       group_by(non_white) %>%
       summarize(count = n()) %>%
-      plot_ly(labels = ~non_white, values = ~count) %>%
+      mutate(non_white = case_when(non_white == 0 ~ "White",
+                                   non_white == 1 ~ "Non-White")) %>%
+      plot_ly(labels = ~non_white, values = ~count,
+              textinfo = "label", # "label+percent"
+              textfont = list(size = 10),
+              marker = list(colors = colors_white)) %>%
       add_pie(hole = 0.5) %>%
       layout(title = "Identity",  
              showlegend = FALSE,
              xaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE),
              yaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE))
     
   })
-
+  
+  # Make a function to wrangle data for by topic bar charts
+  
+  data_bytopic <- reactive({
+    req(input$topic)
+    
+    cnames <- labels$Metric[labels$Topic == input$topic] # substitute with input$topic
+    
+    # Make E + V only for ranking purposes
+    rank_EV <- refine_top_20() %>% # substitute with refine_top_20()
+      select(all_of(cnames)) %>%
+      pivot_longer(cols = everything(),
+                   names_to = "Metric",
+                   values_to = "Response") %>%
+      group_by(Metric, Response) %>%
+      summarize(n = n()) %>%
+      drop_na() %>%
```

Comment:
```suggestion
      count(Metric, Response) %>%
      drop_na() %>%
```

Reply:
I think that does the same thing, but probably doesn't make a big difference in performance